org.opentosca.planbuilder.core.bpel/src/org/opentosca/planbuilder/core/bpel/BPELFreezeProcessBuilder.java

```diff
@@ -0,0 +1,280 @@
+				serviceTemplateId.toString(), definitions.getId(), csarName);
+		return null;
+	}
+
+	@Override
+	public List<AbstractPlan> buildPlans(final String csarName, final AbstractDefinitions definitions) {
+		BPELFreezeProcessBuilder.LOG.info("Builing the Plans");
+		final List<AbstractPlan> plans = new ArrayList<>();
+		for (final AbstractServiceTemplate serviceTemplate : definitions.getServiceTemplates()) {
+			QName serviceTemplateId;
+			// targetNamespace attribute doesn't has to be set, so we check it
+			if (serviceTemplate.getTargetNamespace() != null) {
+				serviceTemplateId = new QName(serviceTemplate.getTargetNamespace(), serviceTemplate.getId());
+			} else {
+				serviceTemplateId = new QName(definitions.getTargetNamespace(), serviceTemplate.getId());
+			}
+
+			BPELFreezeProcessBuilder.LOG.debug("ServiceTemplate {} has no BackupPlan, generating BackuopPlan",
+					serviceTemplateId.toString());
+			final BPELPlan newBuildPlan = buildPlan(csarName, definitions, serviceTemplateId);
+
+			if (newBuildPlan != null) {
+				BPELFreezeProcessBuilder.LOG
+						.debug("Created BackupPlan " + newBuildPlan.getBpelProcessElement().getAttribute("name"));
+				plans.add(newBuildPlan);
+			}
+
+		}
+		return plans;
+	}
+
+	private boolean isDockerContainer(final AbstractNodeTemplate nodeTemplate) {
+		if (nodeTemplate.getProperties() == null) {
+			return false;
+		}
+		final Element propertyElement = nodeTemplate.getProperties().getDOMElement();
+		final NodeList childNodeList = propertyElement.getChildNodes();
+
+		int check = 0;
+		boolean foundDockerImageProp = false;
+		for (int index = 0; index < childNodeList.getLength(); index++) {
+			if (childNodeList.item(index).getNodeType() != Node.ELEMENT_NODE) {
+				continue;
+			}
+			if (childNodeList.item(index).getLocalName().equals("ContainerPort")) {
+				check++;
+			} else if (childNodeList.item(index).getLocalName().equals("Port")) {
+				check++;
+			} else if (childNodeList.item(index).getLocalName().equals("ContainerImage")) {
+				foundDockerImageProp = true;
+			}
+		}
+
+		if (check != 2) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * This Methods Finds out if a Service Template Container a freeze method and
+	 * then creats a freeze plan out of this method
+	 *
+	 * @param plan            the plan to execute the plugins on
+	 * @param serviceTemplate the serviceTemplate the plan belongs to
+	 * @param propMap         a PropertyMapping from NodeTemplate to Properties to
+	 *                        BPELVariables
+	 */
+	private List<BPELScopeActivity> runPlugins(final BPELPlan plan, final PropertyMap propMap) {
+
+		final List<BPELScopeActivity> changedActivities = new ArrayList<>();
+
+		for (final BPELScopeActivity templatePlan : plan.getTemplateBuildPlans()) {
+			if (templatePlan.getNodeTemplate() != null) {
+
+				// Only looks at Nodes that are no docker engine
+				if (!org.opentosca.container.core.tosca.convention.Utils
```

Comment:
Why do you need this here?

Reply:
The Docker Engine is the DinD Container. It's not good for the security if i allow to run scripts on this.
The script could mess with the other deployed Services